cmd/cli/main.go

```diff
@@ -115,24 +115,26 @@ func main() {
 						}
 					}
 
+					runner.EndRun()
```

Comment:
I was looking into the code of `EndRun`. Outside of the scope of the PR but could you remove the `if` conditional it has inside. It is usually not a good practice to have a conditional inside a function, it's better to have it outside. For two reasons:
* Is easy for people to understand that `EndRun` only makes sense in Proof Mode
* You remove the performance cost of making a call that does nothing, when there is not proof mode. (Although, in this case, the function is probably getting inlined).

cc: @MaksymMalicki